core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -3472,6 +3483,300 @@ protected UniqueIDPaginatedSearchResult doGetUserListWithID(Condition condition,
+                    DatabaseUtil.udpateUserRoleMappingInBatchMode(dbConnection, sqlStmt2, userIdList, groupName);
+                }
+            }
+            dbConnection.commit();
+        } catch (SQLException e) {
+            DatabaseUtil.rollBack(dbConnection);
+            String msg = "Error occurred while adding group : " + groupName;
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
+            }
+            throw new UserStoreException(msg, e);
+        } catch (Exception e) {
+            String errorMessage = "Error occurred while adding group : " + groupName;
+            if (log.isDebugEnabled()) {
+                log.debug(errorMessage, e);
+            }
+            if (e instanceof UserStoreException && ERROR_CODE_DUPLICATE_WHILE_WRITING_TO_DATABASE.getCode()
+                    .equals(((UserStoreException) e).getErrorCode())) {
+                // Duplicate entry.
+                throw new UserStoreException(errorMessage, ERROR_CODE_DUPLICATE_WHILE_ADDING_ROLE.getCode(), e);
+            }
+            // Other SQL Exceptions.
+            throw new UserStoreException(errorMessage, e);
+        } finally {
+            DatabaseUtil.closeAllConnections(dbConnection);
+        }
+    }
+
+    @Override
+    public void doUpdateGroupNameByGroupId(String groupId, String newGroupName) throws UserStoreException {
+
+        if (!isUniqueGroupIdEnabled()) {
+            throw new UserStoreException("Group ID is not supported for userstore: " + getMyDomainName());
+        }
+        if (StringUtils.isBlank(groupId)) {
+            throw new UserStoreClientException(ERROR_EMPTY_GROUP_ID.getMessage());
+        }
+        if (StringUtils.isBlank(newGroupName)) {
+            throw new UserStoreClientException(ERROR_EMPTY_GROUP_NAME.getMessage());
+        }
+        String currentGroupName = doGetGroupNameFromGroupId(groupId);
+        if (StringUtils.isBlank(currentGroupName)) {
+            throw new UserStoreClientException(String.format(ERROR_NO_GROUP_FOUND_WITH_ID.getMessage(), groupId,
+                    CarbonContext.getThreadLocalCarbonContext().getTenantDomain()));
+        }
+
+        String domainFreeCurrentGroupName = UserCoreUtil.removeDomainFromName(currentGroupName);
+        String domainFreeNewGroupName = UserCoreUtil.removeDomainFromName(newGroupName);
+        if (!StringUtils.equalsIgnoreCase(domainFreeCurrentGroupName, domainFreeNewGroupName) &&
+                isExistingRole(domainFreeNewGroupName)) {
+            throw new UserStoreException("Role name: " + domainFreeNewGroupName
+                    + " in the system. Please pick another role name.");
+        }
+        String sqlStmt = realmConfig.getUserStoreProperty(JDBCRealmConstants.UPDATE_GROUP_NAME);
+        if (StringUtils.isBlank(sqlStmt)) {
+            throw new UserStoreException("The sql statement for update role name is null");
+        }
+        Connection dbConnection = null;
+        try {
```

Comment:
Can we use try with resources?

Reply:
Addressed with https://github.com/wso2/carbon-kernel/pull/3847/commits/344924abcc35406833451b3a6c68ab68ec225912